src/Components/test/testassets/TestServer/Pages/Authentication.cshtml

```diff
@@ -0,0 +1,72 @@
+@page
+@using Microsoft.AspNetCore.Authentication
+@using System.Security.Claims
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Authentication</title>
+</head>
+<body>
+    <h1>Authentication</h1>
+    <p>
+        This is a completely fake login mechanism for automated test purposes.
+        It accepts any username, with no password.
+    </p>
+    <p>
+        <strong>Obviously you should not do this in real applications.</strong>
+        See also: <a href="https://docs.microsoft.com/aspnet/core/security">documentation on configuring a real login system.</a>
+    </p>
+
+    <fieldset>
+        <h3>Sign in</h3>
+
+        @* Do not use method="get" for real login forms. This is just to simplify E2E tests. *@
+        <form method="get">
+            <p>
+                User name:
+                <input name="username" />
+            </p>
+            <p>
+                <button type="submit">Submit</button>
+            </p>
+        </form>
+    </fieldset>
+
+    <fieldset>
+        <h3>Status</h3>
+        <p>
+            Authenticated: <strong>@User.Identity.IsAuthenticated</strong>
+            Username: <strong>@User.Identity.Name</strong>
+        </p>
+        <a href="Authentication?signout=true">Sign out</a>
+    </fieldset>
+</body>
+</html>
+
+@functions {
+    public async Task<IActionResult> OnGet()
+    {
+        if (Request.Query["signout"] == "true")
+        {
+            await HttpContext.SignOutAsync();
+            return Redirect("Authentication");
+        }
+
+        var username = Request.Query["username"];
+        if (!string.IsNullOrEmpty(username))
+        {
+            var claims = new List<Claim>
+            {
+                new Claim(ClaimTypes.Name, username),
+                new Claim("test-claim", "Test claim value"),
+            };
+
+            await HttpContext.SignInAsync(
+                new ClaimsPrincipal(new ClaimsIdentity(claims, "FakeAuthenticationType")));
```

Comment:
Should we use the same scheme here? `CookieAuthenticationDefaults.AuthenticationScheme` - not sure that it matters.

Reply:
I slightly prefer having specifying `FakeAuthenticationType` here because it clarifies that we're indifferent to the scheme name. But overall I agree it doesn't matter much!